Foundation/Host.swift

```diff
@@ -38,8 +38,8 @@ open class Host: NSObject {
     static internal func currentHostName() -> String {
         let hname = UnsafeMutablePointer<Int8>.allocate(capacity: Int(NI_MAXHOST))
         defer {
-            hname.deinitialize()
-            hname.deallocate(capacity: Int(NI_MAXHOST))
+            hname.deinitialize(count: 1)
```

Comment:
I've always found this method confusing.  What does it actually _do_?

https://developer.apple.com/documentation/swift/unsafemutablepointer/1641720-deinitialize is no help, it just says that `deinitialize()` deinitializes!  Does it zero the memory, for example?  I assume not otherwise the documentation would say so.

If it is just changing the "state" of the memory then I question the meaning/utility of the "states".

Reply:
@ianpartridge Isn't `deinitialize` for objects where the `deinit` method is called on them? If Im correct its not needed for arrays of `Int8`

Reply:
No I think this is unrelated to `deinit` which only applies to classes. `UnsafeMutablePointer` et al are structs.

Reply:
I was referring to the type stored in the allocated memory not the `UnsafeMutablePointer` itself.

Reply:
Oh I see what you mean, sorry :) If that's correct then the doc doesn't mention it!

Reply:
deinitialize does nothing for trivial types, it doesn’t actually need to be there

Reply:
Oh OK! Could you update this PR to remove it then?

Reply:
I’ve removed those calls in `Host.swift` though I would hold off removing them elsewhere because it could inadvertently introduce memory bugs if the type isn’t absolutely a trivial type and will never be anything but a trivial type in the future

Reply:
I’ve removed those calls in `Host.swift` though I would hold off removing them elsewhere because it could inadvertently introduce memory bugs if the type isn’t absolutely a trivial type and will never be anything but a trivial type in the future